### PR TITLE
feat: convert project to monorepo, setup changesets

### PR DESCRIPTION
**Pull request by web-padawan**
## What I did

1. Restructured project as a monorepo using yarn workspaces (see moved files below)
2. Configured `changesets` to bump versions / CHANGELOG files, entered `1.0.0-pre`

```
 51 files changed, 477 insertions(+), 299 deletions(-)
 create mode 100644 packages/api-common/package.json
 create mode 100644 packages/api-common/src/index.ts
 rename src/api-viewer-mixin.ts => packages/api-common/src/manifest-mixin.ts (79%)
 rename {src/lib => packages/api-common/src}/manifest.ts (100%)
 rename {src/styles => packages/api-common/src}/shared-styles.ts (100%)
 rename src/lib/utils.ts => packages/api-common/src/templates.ts (100%)
 create mode 100644 packages/api-common/src/utils.ts
 create mode 100644 packages/api-common/tsconfig.json
 rename {src => packages/api-demo/src}/api-demo.ts (69%)
 rename src/api-demo-base.ts => packages/api-demo/src/base.ts (92%)
 rename {src => packages/api-demo/src}/controllers/events-controller.ts (92%)
 rename {src => packages/api-demo/src}/controllers/slots-controller.ts (92%)
 rename {src => packages/api-demo/src}/controllers/styles-controller.ts (93%)
 rename src/api-viewer-demo.ts => packages/api-demo/src/layout.ts (94%)
 rename src/styles/demo-styles.ts => packages/api-demo/src/styles.ts (72%)
 rename src/lib/demo-controls.ts => packages/api-demo/src/ui/controls.ts (95%)
 rename src/lib/demo-events.ts => packages/api-demo/src/ui/events.ts (100%)
 rename {src/lib => packages/api-demo/src/ui}/highlight-css.ts (100%)
 rename {src/lib => packages/api-demo/src/ui}/knobs.ts (96%)
 rename {src/lib => packages/api-demo/src/ui}/renderer.ts (99%)
 rename src/lib/demo-snippet.ts => packages/api-demo/src/ui/snippet.ts (97%)
 rename {src => packages/api-docs/src}/api-docs.ts (61%)
 rename src/api-docs-base.ts => packages/api-docs/src/base.ts (89%)
 rename src/api-viewer-docs.ts => packages/api-docs/src/layout.ts (94%)
 rename src/styles/docs-styles.ts => packages/api-docs/src/styles.ts (98%)
 rename {src/lib => packages/api-docs/src/utils}/markdown.ts (100%)
 create mode 100644 packages/api-tabs/package.json
 rename {src => packages/api-tabs/src}/api-viewer-panel.ts (100%)
 rename {src => packages/api-tabs/src}/api-viewer-tab.ts (100%)
 rename {src => packages/api-tabs/src}/api-viewer-tabs.ts (98%)
 create mode 100644 packages/api-tabs/tsconfig.json
 rename {src => packages/api-viewer/src}/api-viewer.ts (76%)
 rename src/api-viewer-base.ts => packages/api-viewer/src/base.ts (88%)
 rename src/styles/viewer-styles.ts => packages/api-viewer/src/styles.ts (55%)
 delete mode 100644 src/styles/highlight-styles.ts
 ```

### CODE REVIEW

File: packages/api-common/src/manifest-mixin.ts [hunk 1]
```diff
@@ -19,10 +19,10 @@ export const emptyDataWarning = html`
   <div part="warning">No custom elements found in the JSON file.</div>
 `;
 
-export const ApiViewerMixin = <T extends Constructor<LitElement>>(
+export const ManifestMixin = <T extends Constructor<LitElement>>(
```
Comment (bennypowers) [inline packages/api-common/src/manifest-mixin.ts:22]: Should we prefer this to be a controller?

If so we can also defer that change to another PR

Comment (web-padawan) [reply packages/api-common/src/manifest-mixin.ts:22]: This mixin defines some properties so I'm not sure how to handle it with controller.

Comment (bennypowers) [reply packages/api-common/src/manifest-mixin.ts:22]: can we move those to setters on the controller?

Comment (bennypowers) [reply packages/api-common/src/manifest-mixin.ts:22]: or have the controller read host attrs for initial setup state?

Comment (bennypowers) [reply packages/api-common/src/manifest-mixin.ts:22]: we could have the controller constructor receive an adapter object with methods that translate between controller state and host state, perhaps

Comment (web-padawan) [reply packages/api-common/src/manifest-mixin.ts:22]: Feel free to create an issue if you think it's worth to get rid of the mixin. I have plans to change some components from Lit to vanilla in the future, so this code will need to be updated acordingly to remove `@property`.

**Pull request merged**